frontend/src/packages/api/types/filesystem.ts

```diff
@@ -0,0 +1,30 @@
+// FilesystemEntry is the contract for the type of response we receive from the backend
+export type FilesystemEntry = {
+    EntityID: string,
+    EntityName: string,
+    IsDocument: boolean,
+    Parent: string,
+    Children: FilesystemEntry[]
+}
+
+export type CreateFilesystemEntryResponse = {
+    NewID: string
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const hasFieldOfType = (o: any, fieldName: string, type: string): boolean =>
+    fieldName in o && (typeof o[fieldName]) === type;
```

Comment:
not sure if it makes more sense to use `fieldType` here since you're using `fieldName` as well - i got really confused here for a minute because i thought `type` was some sort of built-in, or some typescript magic was going on here 😳